lib/hanami/router.rb

```diff
@@ -637,7 +639,7 @@ def not_allowed(env)
     # @since 2.0.0
     # @api private
     def not_found
-      [404, { "Content-Length" => "9" }, ["Not Found"]]
+      ->(_) { [404, { "Content-Length" => "9" }, ["Not Found"]] }
     end
```

Comment:
This could be a constant, I think, but haven't seen any other constants defined and left it as method.

Reply:
@katafrakt It used to be a constant, but it caused a bug: https://github.com/hanami/router/pull/201

If we wrap the array into a lambda then, it's safe to assign the lambda to a constant, as each time it will be invoked, it will instantiate a new array, so we won't recreate the bug above.

```ruby
NOT_FOUND = ->(_) { [404, { "Content-Length" => "9" }, ["Not Found"]] }.freeze
```